### PR TITLE
search-backend-module-elasticsearch: refactor + remove options

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

Work towards removing options from backend modules. The index templates were moved to config but customizing the translator requires a custom module.

@backstage/discoverability-maintainers PTAL 🙏 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/popular-beds-invent.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'@backstage/plugin-search-backend-module-elasticsearch': patch
+---
+
+Index templates can now be configured through configuration under the `search.elasticsearch.indexTemplates`. In addition, the `ElasticSearchSearchEngine.fromConfig` now also accepts a `LoggerService` as the `logger` option as well as a new `translator` option.
```
Comment (freben) [inline .changeset/popular-beds-invent.md:5]: Hm unfortunate that this wasn't already under a `search.engines` prefix or so. But that can always be added as an alias some time down the line of course

File: plugins/search-backend-module-elasticsearch/src/alpha.ts [hunk 2]
```diff
@@ -17,58 +17,65 @@ import {
...
- * @alpha
  * Search backend module for the Elasticsearch engine.
+ *
+ * @alpha
+ * @remarks
+ *
+ * If you need to customize the search engine beyond what is supported by the
+ * static configuration, you can create a custom module using the `ElasticSearchSearchEngine`.
+ *
+ * For example, this lets you configure your own query translator:
```
Comment (freben) [inline plugins/search-backend-module-elasticsearch/src/alpha.ts:32]: ⭐ but, should it be a section in the README or even in the docs instead?

**Review by freben — APPROVED** [review]

**Comment by Rugvip** [comment]
Needs a refactor following #19337

File: .changeset/popular-beds-invent.md (hunk 1 above)
Comment (Rugvip) [reply .changeset/popular-beds-invent.md:5]: Yep, will leave that for someone else 😁

**Review by camilaibs — APPROVED** [review]

**Pull request merged**